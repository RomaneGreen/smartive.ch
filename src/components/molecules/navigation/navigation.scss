@import "../../../scss/colors";
@import "../../../scss/responsive";

.navigation {
  @media #{$large-up} {
    box-shadow: none;
  }

  ul {
    list-style: none;
    display: flex;
    align-content: center;
    margin: .5rem 0;

    @media #{$large-up} {
      margin: 1rem 0;
    }

    li {
      display: inline-block;
      position: relative;
      text-align: center;
      flex-grow: 1;
      padding: 0;

      @media #{$large-up} {
        text-align: right;

        &:last-of-type a {
          margin-right: 0;
        }
      }
    }

    a {
      margin: 0;
      padding: 1rem 0;
      text-decoration: none;
      font-size: .75rem;
      color: $smartive-dark-blue;
      border-bottom: .125rem solid transparent;

      transition: border-bottom-color .2s ease-in-out;

      &.is-active,
      &:hover {
        color: $smartive-blue;

        @media #{$large-up} {
          border-bottom: .125rem solid $smartive-blue;
          color: $smartive-dark-blue;
        }
      }

      @media #{$large-up} {
        font-size: 1.2rem;
        padding: .5rem 0;
        margin: .5rem 1rem;
      }
    }
  }

  .navigation-item--medium {
    &:hover {
      border-bottom: .125rem solid $smartive-dark-blue;
    }

    &:after {
      display: none;

      @media #{$large-up} {
        position: absolute;
        display: inline;
        height: 1.5rem;
        width: 2rem;
        background-image: url('./medium-monogram.svg');
        background-size: contain;
        background-repeat: no-repeat;
        background-position: right center;
        top: .4rem;
        content: ' ';
        opacity: 0;
        transition: opacity .2s ease-in-out;
      }
    }

    &:hover:after {
      opacity: 1;
    }
  }
}
