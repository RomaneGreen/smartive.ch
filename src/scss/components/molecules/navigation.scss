@import "../../colors";

.navigation {
  @media #{$large-up} {
    box-shadow: none;
  }

  ul {
    list-style: none;
    display: flex;
    align-content: center;
    margin: .5rem 0;

    @media #{$large-up} {
      margin: 1rem 2.5rem 1rem 0;
    }

    @media #{$xlarge-up} {
      margin: 1rem 1.5rem 1rem 0;
    }

    @media #{$xxlarge-up} {
      margin: 1rem .25rem 1rem 0;
    }

    li {
      display: inline-block;
      position: relative;
      text-align: center;
      flex-grow: 1;
      padding: 0;

      &:first-of-type a {
        margin-left: 0;
      }
    }

    a {
      margin: 0;
      padding: 1rem 0;
      text-decoration: none;
      font-size: .75rem;
      color: $smartive-dark-blue;
      border-bottom: .125rem solid transparent;

      transition: border-bottom-color .2s ease-in-out;

      &.is-active,
      &:hover {
        color: $smartive-blue;

        @media #{$large-up} {
          border-bottom: .125rem solid $smartive-blue;
          color: $smartive-dark-blue;
        }
      }

      @media #{$large-up} {
        font-size: 1.2rem;
        padding: .5rem 0;
        margin: .5rem 1rem;
      }
    }
  }

  .navigation-item--medium {
    border-bottom: none !important;
    margin-right: 0;

    &:hover {
      border-bottom: none !important;
    }

    &:before {
      width: 75px;

      @media #{$large-up} {
        position: absolute;
        content: ' ';
        height: .125rem;
        bottom: -.25rem;
        background: transparent;

        transition: background-color .2s ease-in-out;
      }
    }

    &:hover:before {
      background-color: $smartive-blue;
    }

    &:after {
      display: none;

      @media #{$large-up} {
        position: absolute;
        display: inline;
        height: 2.25rem;
        width: 2.25rem;
        padding-left: .5rem;
        margin-right: 0;
        background-image: url('./medium-m-dark-100px.png');
        background-size: contain;
        background-repeat: no-repeat;
        background-position: right center;
        top: 0;
        content: ' ';
      }
    }
  }
}
