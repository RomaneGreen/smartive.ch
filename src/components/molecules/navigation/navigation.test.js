import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Navigation } from './navigation';

// This test uses the Shallow Renderer (only one JSX level deep) because the
// underlying gatsby-link relies on the router being passed in the context.
// Therefore this test doesn't test the link implementation.
const renderer = new ShallowRenderer();

describe('Navigation', () => {
  it('renders correctly', () => {
    const tree = renderer.render(<Navigation />);
    expect(tree).toMatchSnapshot();
  });
});
