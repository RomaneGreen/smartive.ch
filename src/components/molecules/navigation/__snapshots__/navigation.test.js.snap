// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  className="navigation"
>
  <ul>
    <li>
      <GatsbyLink
        activeClassName="is-active"
        to="/leistungen"
      >
        Leistungen
      </GatsbyLink>
    </li>
    <li>
      <GatsbyLink
        activeClassName="is-active"
        to="/projekte"
      >
        Projekte
      </GatsbyLink>
    </li>
    <li>
      <GatsbyLink
        activeClassName="is-active"
        to="/agentur"
      >
        Agentur
      </GatsbyLink>
    </li>
    <li>
      <GatsbyLink
        activeClassName="is-active"
        to="/kontakt"
      >
        Kontakt
      </GatsbyLink>
    </li>
    <li>
      <a
        className="navigation-item--medium"
        href="https://blog.smartive.ch"
        rel="noopener noreferrer"
        target="_blank"
      >
        Blog
      </a>
    </li>
  </ul>
</nav>
`;
